### ISSUE TITLE
WASM compatibility

### ISSUE THREAD

**Issue by stuartlynn** [kylebarron/geopolars#63]
We should keep an eye on this as we go. Currently there is some work needed on the Polars side to support this.

As an experiment I managed to get the code compiling for WASM here https://github.com/stuartlynn/geopolars/tree/wasm_test

This uses a branch of Polars ( https://github.com/stuartlynn/polars/tree/wasm_rebase) which incorporates some commits from @dimasukr to get poplars to compile to  WASM .

Haven't done much testing yet but everything builds ok

**Comment by stuartlynn** [comment]
Yeah 100% agree. The only real difference on the geopolars side just now is a few different dependencies, so we could also for now just set up a wasm feature with those deps

**Comment by kylebarron** [comment]
I think it would be cleanest in a new directory, rather than as a new feature within the `geopolars` directory

**Comment by stuartlynn** [comment]
Ah ok sorry I think I misunderstood. You mean create a new submodule for WASM support and import the core geopolars modules (with appropriate flags) similar to how we are doing it with python?

**Comment by justin0mcateer** [comment]
It looks like Polars has basic support for targeting WASM now:
https://github.com/pola-rs/polars/pull/6050

Is that still a goal for this project?

I saw some discussion of using/wrapping GEOS for some operations. From what I understand, it seems like that would make targeting WASM much more difficult.

**Comment by kylebarron** [comment]
Any GEOS integration will definitely stay optional so that it can be excluded from wasm builds

**Comment by deanm0000** [comment]
fyi there's this https://github.com/pola-rs/js-polars

**Comment by kylebarron** [comment]
I'm aware, but there's not really an avenue to connect to it in the near term. The story for dynamic linking/extensions in the browser is not great (although duckdb-wasm managed to do it).

In theory we could have a static build that includes all js-polars methods, but as of now js-polars really isn't maintained (it doesn't even depend on a released version). So Python will definitely be the primary target of geopolars in the short and medium term.

**Comment by deanm0000** [comment]
Yeah duckdb gets a lot of benefit from having the vast majority of their api in sql statements that are just strings.

**Comment by kylebarron** [comment]
Closing as stale. The priority of this project is on the Python bindings, and I'm inclined to only consider WASM compatibility once Python bindings are very stable.
